src/todo.js

```diff
@@ -8,98 +9,96 @@ export default class TodoTasks {
         document.querySelector(`.content${i}`).classList.add('completed');
+        this.taskList[i].completed = true;
+        localStorage.setItem('tasks', JSON.stringify(this.taskList));
+      } else {
+        document.querySelector(`.content${i}`).classList.remove('completed');
+        this.taskList[i].completed = false;
+        localStorage.setItem('tasks', JSON.stringify(this.taskList));
       }
+    });
+  }
 
-      checkbox.addEventListener('change', () => {
-        if (checkbox.checked) {
-          document.querySelector(`.content${i}`).classList.add('completed');
-          this.taskList[i].completed = true;
-          localStorage.setItem('tasks', JSON.stringify(this.taskList));
-        } else {
-          document.querySelector(`.content${i}`).classList.remove('completed');
-          this.taskList[i].completed = false;
-          localStorage.setItem('tasks', JSON.stringify(this.taskList));
-        }
+  removeTask = (tasks, taskTrashBtns) => {
+    for (let i = 0; i < tasks.length; i += 1) {
+      taskTrashBtns[i].addEventListener('click', () => {
+        this.taskList.splice(tasks[i].getAttribute('id'), 1);
+        this.taskList.sort((a, b) => a.index - b.index);
+        this.resetIndex();
+        localStorage.setItem('tasks', JSON.stringify(this.taskList));
+        this.showTodoList();
       });
     }
+  }
 
-    removeTask = (tasks, taskTrashBtns) => {
-      for (let i = 0; i < tasks.length; i += 1) {
-        taskTrashBtns[i].addEventListener('click', () => {
-          this.taskList.splice(tasks[i].getAttribute('id'), 1);
-          this.taskList.sort((a, b) => a.index - b.index);
-          this.resetIndex();
-          localStorage.setItem('tasks', JSON.stringify(this.taskList));
-          this.showTodoList();
-        });
-      }
-    }
-
-    saveOnChangeTask = (taskDescriptions) => {
-      taskDescriptions.forEach((t) => {
-        t.addEventListener('change', () => {
-          this.taskList[t.getAttribute('id')].description = t.value;
-          localStorage.setItem('tasks', JSON.stringify(this.taskList));
-        });
+  saveOnChangeTask = (taskDescriptions) => {
+    taskDescriptions.forEach((t) => {
+      t.addEventListener('change', () => {
+        this.taskList[t.getAttribute('id')].description = t.value;
+        localStorage.setItem('tasks', JSON.stringify(this.taskList));
       });
-    }
+    });
+  }
 
-    getLocalStorage = () => this.taskList;
+  getLocalStorage = () => this.taskList;
 
-    emptyList = () => {
-      this.taskList.forEach((task) => {
-        if (task.completed) {
-          this.taskList.splice(task.index, 1);
-          this.taskList.sort((a, b) => a.index - b.index);
-          this.resetIndex();
-        }
-      });
-      localStorage.setItem('tasks', JSON.stringify(this.taskList));
-      this.showTodoList();
-    }
+  clearAllCompleted = () => {
+    this.taskList = this.taskList.filter((task) => {
+      if (!task.completed) { return task; }
+    });
```

Comment:
- [ ] For the implementation here,  'filter()' method is incomplete. It should either explicitly return true to keep the task or omit the if statement altogether since the 'filter()' function keeps elements for which the callback function returns a truthy value. The corrected code could be:

`this.taskList = this.taskList.filter((task) => !task.completed);`